### ISSUE TITLE
NS4000 Unexpected warning in foreach loop

### ISSUE THREAD

**Issue by appel1** [nsubstitute/NSubstitute.Analyzers#93]
NSubstitute 4.0.0.0
NSubstitute.Analyzers.CSharp 1.0.9

Perhaps I'm missing something but this looks like a false positive to me. 

```csharp
using System.Linq;
using NSubstitute;

namespace ClassLibrary3
{
    public class Class1
    {
        public void Foo(IFoo foo, IEnumerable<Description> descriptions)
        {
            foreach (Description d in descriptions)
            {
                foo.Value.Returns(d.Value); // <- Warning	NS4000	Returns() is set with a method that itself calls Returns. This can cause problems with NSubstitute. Consider replacing with a lambda: Returns(x => d.Value).
            }
        }
    }

    public class Description
    {
        public int Value { get; set; }
    }

    public interface IFoo
    {
        int Value { get; }
    }
}
```

**Comment by dtchepak** [comment]
Thanks for raising this! The code looks fine to me. I think you are correct that it is a false positive.

**Comment by tpodolak** [comment]
Published in [1.0.10](https://www.nuget.org/packages/NSubstitute.Analyzers.CSharp/1.0.10)
